### PR TITLE
Introduction notebooks

### PR DESCRIPTION
**Pull request by christian512**
Adds notebooks from the Introduction Section on the QuTiP website.

### CODE REVIEW

File: tutorials-v4/python-introduction/Beginning_Python.md [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+
+$$N_{\rm c}=N\pm \epsilon, \ \ \forall~|\epsilon|< \epsilon_{\rm m}.$$
+
+**Take Home Message** - All double-precision decimal numbers that are not factors of two will have error in the 15th decimal place.  This can lead to errors in your numerical solutions if you are not careful.
+
+
+## Making Python Smarter Using NumPy
+
+
+Python itself as limited support for mathematics outside of simple arithmetic.  Therefore, we will use the functions in the NumPy module to do more impressive, and faster, calculations.  We have imported NumPy already at the top of this notebook and can use it now by referring to `np`.
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/Beginning_Python.md:109]:
```suggestion
Python itself has limited support for mathematics outside of simple arithmetic.  Therefore, we will use the functions in the NumPy module to do more impressive, and faster, calculations.  We have imported NumPy already at the top of this notebook and can use it now by referring to `np`.
```

File: tutorials-v4/python-introduction/Beginning_Python.md [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+
+```python
+np
+```
+
+Here we are asking Python to get all of the basic functions (this is what * means) from the NumPy module.  We can now do more impressive calculations:
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/Beginning_Python.md:115]:
```suggestion
We can now do more impressive calculations:
```

I guess some of this was old text?

File: tutorials-v4/python-introduction/Beginning_Python.md [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+
+What happens if you try to use a variable without first defining it? 
+
+Python would give us an error that the variable is not defined.  In addition, there are several words that are reserved by the Python language and cannot be used as variables:
+
+    and, as, assert, break, class, continue, def, del, elif, else, except, 
+    exec, finally, for, from, global, if, import, in, is, lambda, not, or,
+    pass, print, raise, return, try, while, with, yield
+    
+Other than the above reserved words, your variables can be anything that starts with a letter or the underscore character "$\_$" preceded by any combination of alphanumeric characters and "$\_$".  Note that using upper or lower case letters will give you two different variables.
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/Beginning_Python.md:161]:
```suggestion
Other than the above reserved words, your variables can be anything that starts with a letter or the underscore character "$\_$" followed by any combination of alphanumeric characters and "$\_$".  Note that using upper or lower case letters will give you two different variables.
```

File: tutorials-v4/python-introduction/Beginning_Python.md [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+spring_constant = np.sqrt(2 / 5)
+```
+
+It is also good practice to use variable names that correspond to the physical quantity that the variable represents.
+
+
+## Strings
+
+
+Often times we want to print some text along with our variables, ask the user for input, or actually use the words and letters themselves as variables (e.g. in DNA analysis).  All of these can be accomplished using **strings**.  We have already seen one string already in this class:
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/Beginning_Python.md:186]:
```suggestion
Often we want to print some text along with our variables, ask the user for input, or actually use the words and letters themselves as variables (e.g. in DNA analysis).  All of these can be accomplished using **strings**.  We have already seen one string already in this class: 
```

File: tutorials-v4/python-introduction/Beginning_Python.md [hunk 5]
```diff
@@ -0,0 +1,681 @@
...
+
+```python
+"Hello Class"
+```
+
+We can also use double quotes
+
+```python
+"Hello Class"
+```
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/Beginning_Python.md:196]:
```suggestion
We can also use single quotes, e.g. `'Hello Class'`.
```
I guess some auto formatting changed the example above from single quotes to double?

File: tutorials-v4/python-introduction/NumPy_Array_Basics.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+
+
+## Introduction
+
+<!-- #region -->
+Until now we have been using lists as a way of storing multiple elements together.  However, when doing numerical computations, lists are not very good.  For example, what if I wanted to add one to a list of numbers?  We couldn't write
+
+```python
+a=[1,2,3]
+a=a+1
```
Comment (hodgestar) [inline tutorials-v4/python-introduction/NumPy_Array_Basics.md:34]: I expected these lines to have been formatted to match flake8, but somehow they are not. Do you know why they weren't?

**Review by hodgestar — CHANGES_REQUESTED** [review]
Looks good. I left a few questions. In particular, I'm curious about where the `#region` blocks come from and why they seem to prevent the reformatting of the code blocks inside them.

I only left comments on the tutorials-v4 versions. I guess they should be applied to tutorials-v5 too.

File: tutorials-v4/python-introduction/NumPy_Array_Basics.md (hunk 6 above)
Comment (christian512) [reply tutorials-v4/python-introduction/NumPy_Array_Basics.md]: As you mentioned in the comment above, this is probably due to the `#region` tags. I saw these when I used PyCharm to edit Jupyter Notebooks. Maybe someone did the same thing before..
But I don't know why `black` and `flake8` seem to skip this section.

**Comment by christian512** [comment]
Thanks for the review, @hodgestar! I have applied all the changes and also moved the `images` folder into the `python-introduction` directory.

**Review by hodgestar — APPROVED** [review]
It looks like the tests are picking up some minor issues now -- woot.

I'm happy for this to be merged as soon as the tests pass.

**Pull request merged**